opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -70,9 +70,15 @@ def shutdown(self) -> None:
         """Called when a :class:`opentelemetry.sdk.trace.Tracer` is shutdown.
         """
 
-    def force_flush(self) -> None:
+    def force_flush(self, timeout_millis: int = 30000) -> bool:
         """Export all ended spans to the configured Exporter that have not
         yet been exported.
+
+        Args:
+            timeout_millis: The maximum amount of time to wait for spans to be exported.
+
+        Returns:
+            False if the timeout is exceeded, True otherwise.
```

Comment:
Any thoughts around returning a boolean here rather than raising an exception if the timeout is exceeded?

Reply:
Right, I updated to raise a `RuntimeError` exception. I'm not sure about the type of this exception, there is a [timeout exception on Python](https://docs.python.org/3/library/exceptions.html#TimeoutError) but the description doesn't fully align with our case: "Raised when a system function timed out at the system level. Corresponds to errno ETIMEDOUT."

Reply:
(see https://github.com/open-telemetry/opentelemetry-python/pull/397#discussion_r375821150)